drivers/tee/optee/supp.c

```diff
@@ -88,10 +88,15 @@ u32 optee_supp_thrd_req(struct tee_context *ctx, u32 func, size_t num_params,
 {
 	struct optee *optee = tee_get_drvdata(ctx->teedev);
 	struct optee_supp *supp = &optee->supp;
-	struct optee_supp_req *req = kzalloc(sizeof(*req), GFP_KERNEL);
-	bool interruptable;
+	struct optee_supp_req *req = NULL;
 	u32 ret;
 
+	if (!supp->ctx) {
+		pr_err("There's no tee-supplicant available, please start it\n");
+		return TEEC_ERROR_COMMUNICATION;
```

Comment:
This doesn't deal with the case during boot where tee-supplicant may not have started yet so clients may get unexpected errors with no clear way of dealing with them.

Reply:
If tee-supplicant have not started yet,  it also show this error message.
And ca open session failed with error code 0xffff000e(TEE_ERROR_COMMUNICATION).

Reply:
Yes, but what's the client supposed to do about that? Try again and again and hope that an unspecific error goes away?

Reply:
You are right, close it.